diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index 2c42918..b00b2da 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -1584,6 +1584,16 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
     }
   }
 
+  // by-pass deinterlacer for h264 25fps
+  if (am_private->video_codec_id == AV_CODEC_ID_H264 && am_private->video_rate >= 3840 && am_private->video_rate <= 3844)
+  {
+    CLog::Log(LOGDEBUG, "CAMLCodec::OpenDecoder h264 25fps detected: setting di bypass_all=1");
+    SysfsUtils::SetString("/sys/module/di/parameters/bypass_all", "1");
+  } else {
+    CLog::Log(LOGDEBUG, "CAMLCodec::OpenDecoder setting di bypass_all=0");
+    SysfsUtils::SetString("/sys/module/di/parameters/bypass_all", "0");
+  }
+
   // handle orientation
   am_private->video_rotation_degree = 0;
   if (hints.orientation == 90)
@@ -2422,6 +2432,12 @@ void CAMLCodec::SetVideoRect(const CRect &SrcRect, const CRect &DestRect)
   SysfsUtils::SetString("/sys/class/video/axis", video_axis);
   SysfsUtils::SetInt("/sys/class/video/screen_mode", screen_mode);
 
+  // avoid logspam when by-passing deinterlacer for h264 25fps
+  if (am_private->video_codec_id == AV_CODEC_ID_H264 && am_private->video_rate >= 3840 && am_private->video_rate <= 3844)
+    SysfsUtils::SetString("/sys/module/amvideo/parameters/debug_flag", "0");
+  else
+    SysfsUtils::SetString("/sys/module/amvideo/parameters/debug_flag", "1");
+
   // we only get called once gui has changed to something
   // that would show video playback, so show it.
   ShowMainVideo(true);
