diff --git a/xbmc/pvr/windows/GUIWindowPVRRecordings.cpp b/xbmc/pvr/windows/GUIWindowPVRRecordings.cpp
index 6609655..5dcbb0d 100644
--- a/xbmc/pvr/windows/GUIWindowPVRRecordings.cpp
+++ b/xbmc/pvr/windows/GUIWindowPVRRecordings.cpp
@@ -21,6 +21,7 @@
 #include "threads/SingleLock.h"
 #include "utils/URIUtils.h"
 #include "video/windows/GUIWindowVideoNav.h"
+#include "video/VideoLibraryQueue.h"
 
 #include "pvr/PVRGUIActions.h"
 #include "pvr/PVRManager.h"
@@ -98,6 +99,19 @@ bool CGUIWindowPVRRecordingsBase::OnAction(const CAction &action)
       return true;
     }
   }
+  else if (action.GetID() == ACTION_TOGGLE_WATCHED)
+  {
+    CFileItemPtr pItem = m_vecItems->Get(m_viewControl.GetSelectedItem());
+    if (pItem->IsParentFolder())
+      return false;
+
+    if (pItem && (pItem->HasVideoInfoTag() || pItem->m_bIsFolder))
+    {
+      CVideoLibraryQueue::GetInstance().MarkAsWatched(pItem, pItem->m_overlayIcon != CGUIListItem::ICON_OVERLAY_WATCHED);
+      return true;
+    }
+  }
+
   return CGUIWindowPVRBase::OnAction(action);
 }
 
